src/validation.h

```diff
@@ -187,12 +187,48 @@ void UnlinkPrunedFiles(const std::set<int>& setFilesToPrune);
 /** Prune block files up to a given height */
 void PruneBlockFilesManual(int nManualPruneHeight);
 
-/** (try to) add transaction to memory pool
- * plTxnReplaced will be appended to with all transactions replaced from mempool
- * @param[out] fee_out optional argument to return tx fee to the caller **/
-bool AcceptToMemoryPool(CTxMemPool& pool, TxValidationState &state, const CTransactionRef &tx,
-                        std::list<CTransactionRef>* plTxnReplaced,
-                        bool bypass_limits, bool test_accept=false, CAmount* fee_out=nullptr) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
+/**
+* Validation result for a single transaction mempool acceptance.
+* When m_accepted = true, m_replaced_transactions contains a list
+* of replaced transactions and m_base_fees contains the tx fees.
+*/
+struct MempoolAcceptResult {
+    bool m_accepted;
+    CTransactionRef ptx;
+
+    // Valid when m_accepted = false
```

Comment:
Note that m_state.IsValid() holds when m_accpeted = *true*, so this comment could be confusing. I think the comment can just be removed.

Reply:
I now have the comment "Valid when m_accepted = false" for the tx pointer, not for  the TxValidationState